netsync/manager.go

```diff
@@ -1008,7 +1009,17 @@ func (sm *SyncManager) handleHeadersMsg(hmsg *headersMsg) {
 			peer.Disconnect()
 			return
 		}
-
+		// Processing the block headers that are downloaded and if it passes
+		// all the checks, creating a block node that only contains header.
+		err := chain.ProcessBlockHeader(blockHeader)
```

Comment:
If we call this, then we'd also need to get rid of the `checkBlockHeaderContext` call in `checkBlockContext`, which is called from `maybeAcceptBlock` since it's just duplicate check of the same exact thing.

https://github.com/utreexo/utreexod/blob/53dc2e1096418f7286cb29e7d0b679c3129a524a/blockchain/validate.go#L726-L732